crates/sui-framework/packages/sui-framework/sources/table_vec.move

```diff
@@ -24,82 +24,108 @@ module sui::table_vec {
+        t.contents.add(key, e);
     }
 
+    #[syntax(index)]
     /// Return a mutable reference to the `i`th element in the TableVec `t`.
     /// Aborts if `i` is out of bounds.
     public fun borrow_mut<Element: store>(t: &mut TableVec<Element>, i: u64): &mut Element {
-        assert!(length(t) > i, EIndexOutOfBound);
-        table::borrow_mut(&mut t.contents, i)
+        assert!(t.length() > i, EIndexOutOfBound);
+        &mut t.contents[i]
     }
 
     /// Pop an element from the end of TableVec `t`.
     /// Aborts if `t` is empty.
     public fun pop_back<Element: store>(t: &mut TableVec<Element>): Element {
         let length = length(t);
         assert!(length > 0, EIndexOutOfBound);
-        table::remove(&mut t.contents, length - 1)
+        t.contents.remove(length - 1)
     }
 
     /// Destroy the TableVec `t`.
     /// Aborts if `t` is not empty.
     public fun destroy_empty<Element: store>(t: TableVec<Element>) {
         assert!(length(&t) == 0, ETableNonEmpty);
         let TableVec { contents } = t;
-        table::destroy_empty(contents);
+        contents.destroy_empty();
     }
 
     /// Drop a possibly non-empty TableVec `t`.
     /// Usable only if the value type `Element` has the `drop` ability
     public fun drop<Element: drop + store>(t: TableVec<Element>) {
         let TableVec { contents } = t;
-        table::drop(contents)
+        contents.drop()
     }
 
     /// Swaps the elements at the `i`th and `j`th indices in the TableVec `t`.
     /// Aborts if `i` or `j` is out of bounds.
     public fun swap<Element: store>(t: &mut TableVec<Element>, i: u64, j: u64) {
-        assert!(length(t) > i, EIndexOutOfBound);
-        assert!(length(t) > j, EIndexOutOfBound);
+        assert!(t.length() > i, EIndexOutOfBound);
+        assert!(t.length() > j, EIndexOutOfBound);
         if (i == j) { return };
-        let element_i = table::remove(&mut t.contents, i);
-        let element_j = table::remove(&mut t.contents, j);
-        table::add(&mut t.contents, j, element_i);
-        table::add(&mut t.contents, i, element_j);
+        let element_i = t.contents.remove(i);
+        let element_j = t.contents.remove(j);
+        t.contents.add(j, element_i);
+        t.contents.add(i, element_j);
     }
 
     /// Swap the `i`th element of the TableVec `t` with the last element and then pop the TableVec.
     /// This is O(1), but does not preserve ordering of elements in the TableVec.
     /// Aborts if `i` is out of bounds.
     public fun swap_remove<Element: store>(t: &mut TableVec<Element>, i: u64): Element {
-        assert!(length(t) > i, EIndexOutOfBound);
-        let last_idx = length(t) - 1;
-        swap(t, i, last_idx);
-        pop_back(t)
+        assert!(t.length() > i, EIndexOutOfBound);
+        let last_idx = t.length() - 1;
+        t.swap(i, last_idx);
+        t.pop_back()
     }
+
+    #[test]
+    fun test_swap() {
+        let ctx = &mut sui::tx_context::dummy();
```

Comment:
Do we not have a separate `test` file for this?

Reply:
Evidently not?